test/io.jl

```diff
@@ -41,13 +41,18 @@ module TestIO
                    B = 'a':'c',
                    C = ["A", "B", "C"],
                    D = CategoricalArray('a':'c'),
-                   E = NullableCategoricalArray(["A", "B", "C"]),
-                   E = Vector{?Int}(1:3),
-                   F = nulls(3),
-                   G = fill(null, 3))
+                   E = CategoricalArray(["A", "B", null]),
+                   F = Vector{?Int}(1:3),
+                   G = nulls(3),
+                   H = fill(null, 3))
 
-    answer = Sys.WORD_SIZE == 64 ? 0x3bd4a4a099ae8aac : 0x5ecdc4bb
-    @test hash(sprint(printtable, dt)) == answer
+    DRT = CategoricalArrays.DefaultRefType
+    @test sprint(printtable, dt) == """
+                                    "A","B","C","D","E","F","G","H"
+                                    1,"a","A","CategoricalArrays.CategoricalValue{Char,$DRT} 'a'","CategoricalArrays.CategoricalValue{String,$DRT} "A"","1",NULL,NULL
```

Comment:
Looks like we need to fix printing of `CategoricalValue` from `printtable`: it should definitely use the compact printing. Also, `NULL` should be `null`.

BTW, better add a line break before starting the string so that you can drop the excess indentation.

Reply:
The `show` change was introduced in https://github.com/JuliaLang/julia/pull/22981, see discussion there.